tests/pass-dep/libc/pthread-threadname.rs

```diff
@@ -199,4 +199,24 @@ fn main() {
         .unwrap()
         .join()
         .unwrap();
+
+    // Now set the name for a non-existing thread and verify error codes.
+    cfg_if::cfg_if! {
+        if #[cfg(any(target_os = "linux", target_os = "illumos", target_os = "solaris"))] {
```

Comment:
What about macOS?

Reply:
Forgot about it since it allows changing the thread name only from it  will do that, but only for the get method.

Reply:
Sorry I can't parse that sentence.

Reply:
`pthread_setname_np` doesn't accept a thread handle as the first argument like it is for the other platforms, so the function sets the name for the current thread only.  That's why I forgot about a test for `pthread_getname_np`.

Reply:
Then there should be an `unreachable!()` for this error case in the macOS shim.

Reply:
Also seems like `pthread_getname_np` still allows querying other threads on macOS, so please add a test for that.